api/build_config_template.yaml

```diff
@@ -0,0 +1,38 @@
+apiVersion: v1
+kind: Template
+labels:
+  template: topological-inventory-api-build-config
+metadata:
+  name: topological-inventory-api-build-config
+objects:
+- apiVersion: v1
+  kind: BuildConfig
+  metadata:
+    name: topological-inventory-api
+  spec:
+    source:
+      type: Git
+      git:
+        uri: git://github.com/ManageIQ/topological_inventory-api
+        ref: master
```

Comment:
Having the ability to specify the URI, ref, and commit with parameters is handy to deploy a custom BuildConfig -- to test a PR or branch for example.

Reply:
Side note, the variables I've been using are:
```
      git:
        ref: ${SOURCE_REPOSITORY_REF}
        uri: ${SOURCE_REPOSITORY_URL}
        commit: ${SOURCE_REPOSITORY_COMMIT}
```

Good question... at the moment only master gets deployed in 'buildfactory' and we only deploy custom builders into other namespaces when we wanna build a specific branch or PR. But yeah I don't think a `${IMAGESTREAM_TAG}` variable, with a default of 'latest' would hurt :) We may want more tags to live in buildfactory one day. For now, just sticking with master/latest.